src/legacy/core_plugins/data/public/filter/filter_manager/lib/map_range.ts

```diff
@@ -17,49 +17,58 @@
+import { IndexPatterns, IndexPattern } from '../../../index_patterns';
 
+const TYPE = 'range';
 
-function isScriptedRange(filter) {
-  const params = get(filter, ['script', 'script', 'params']);
-  return params && Object.keys(params).find(key => ['gte', 'gt', 'lte', 'lt'].includes(key));
-}
+const getFirstRangeKey = (filter: RangeFilter) => filter.range && Object.keys(filter.range)[0];
+const getRangeByKey = (filter: RangeFilter, key: string) => get(filter, ['range', key]);
+const isScriptedRange = (filter: RangeFilter) => {
+  const params = get(filter, ['script', 'script', 'params'], {});
 
-function getParams(filter, indexPattern) {
-  const isScriptedRangeFilter = isScriptedRange(filter);
-  const type = 'range';
-  const key = isScriptedRangeFilter ? filter.meta.field : Object.keys(filter.range)[0];
-  const params = isScriptedRangeFilter ? filter.script.script.params : filter.range[key];
+  return Boolean(
+    Object.keys(params).find((key: string) => ['gte', 'gt', 'lte', 'lt'].includes(key))
+  );
+};
 
-  let left = has(params, 'gte') ? params.gte : params.gt;
-  if (left == null) left = -Infinity;
+function getParams(filter: RangeFilter, indexPattern?: IndexPattern) {
+  const isScriptedRangeFilter = isScriptedRange(filter);
+  const key: string = (isScriptedRangeFilter ? filter.meta.field : getFirstRangeKey(filter)) || '';
+  const params: any = isScriptedRangeFilter
+    ? get(filter, 'script.script.params')
+    : getRangeByKey(filter, key);
+  const left = params.gte || params.gt || -Infinity;
+  const right = params.lte || params.lt || Infinity;
 
-  let right = has(params, 'lte') ? params.lte : params.lt;
-  if (right == null) right = Infinity;
+  let value = `${left} to ${right}`;
 
   // Sometimes a filter will end up with an invalid index param. This could happen for a lot of reasons,
   // for example a user might manually edit the url or the index pattern's ID might change due to
   // external factors e.g. a reindex. We only need the index in order to grab the field formatter, so we fallback
   // on displaying the raw value if the index is invalid.
-  let value = `${left} to ${right}`;
-  if (indexPattern && indexPattern.fields.byName[key]) {
+  if (key && indexPattern && indexPattern.fields.byName[key]) {
     const convert = indexPattern.fields.byName[key].format.getConverterFor('text');
+
     value = `${convert(left)} to ${convert(right)}`;
   }
 
-  return { type, key, value, params };
+  return { type: TYPE, key, value, params };
 }
 
-export function mapRange(indexPatterns) {
-  return async function (filter) {
+export const mapRange = (indexPatterns: IndexPatterns) => {
+  return async (filter: RangeFilter) => {
     const isScriptedRangeFilter = isScriptedRange(filter);
+
     if (!filter.range && !isScriptedRangeFilter) {
       throw filter;
     }
 
     try {
-      const indexPattern = await indexPatterns.get(filter.meta.index);
+      const indexPattern = await indexPatterns.get(filter.meta.index || '');
```

Comment:
@ppisljar Is empty string valid input for `indexPatterns.get`?

Reply:
I think I can answer, no. I'll fix that